lib/matplotlib/figure.py

```diff
@@ -2378,6 +2378,13 @@ def axes(self):
 
     get_axes = axes.fget
 
+    def _get_cachedRenderer(self, error_if_none=True):
+        # Get the cached renderer, raising an error if it doesn't exist yet
```

Comment:
We already have a public method https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.get_renderer_cache.html `ax.get_render_cache`, do we want to prompt this to be public on the Figure too (or deprecate the one on Axes?  We do not use it internally, it only shows up where it is defined, in an example, and in the source rst).

Reply:
Good question, and I'm not entirely sure what the answer is as I'm not too sure why/if a user would want to get the cached renderer.

Reply:
I think for exactly the reason we are getting it here! 

I can see an argument than all of the renderer cache stuff should be private. It is a cache and there is no good reason for the user to be poking around the cache, we want to be able to replace/clear it/etc at will.  On the other hand, we expose a couple of public functions that take a renderer as an "optional" (but only because fill it in with the cache) argument.  Making renderers is a bit finicky (and if you make it your self no guarantee that it is consistent with anything you care about!).